### PR TITLE
Show diff on settings change

### LINKED ISSUE #240: Only show difference column when the parameters != original parameters

### PR DESCRIPTION
**Pull request by Log234**
# Description
The difference column in Election Overview should always be show if settings are different from the default settings.

## Issues closed
Closes #240

### CODE REVIEW

File: src/utilities/conditionals.ts [hunk 1]
```diff
@@ -6,3 +8,20 @@
...
+   return (
+       settingsPayload.algorithm !== comparison.algorithm ||
+       comparison.firstDivisor !== settingsPayload.firstDivisor ||
+       settingsPayload.electionThreshold !== comparison.electionThreshold ||
+       comparison.districtThreshold !== settingsPayload.districtThreshold ||
+       settingsPayload.levelingSeats !== comparison.levelingSeats ||
+       settingsPayload.districtSeats !== comparison.districtSeats ||
+       comparison.areaFactor !== settingsPayload.areaFactor
+   );
+};
```
Comment (Log234) [inline src/utilities/conditionals.ts:27]: Codacy found an issue: [Unnecessary semicolon.](https://app.codacy.com/gh/Project-Lavinia/Lavinia-client/pullRequest?prid=6071614)

**Comment by Log234** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 5
           

Complexity decreasing per file
==============================
+ src/utilities/conditionals.ts  -1
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/Project-Lavinia/Lavinia-client/pullRequest?prid=6071614&bid=19881573)

File: src/utilities/conditionals.ts [hunk 2]
```diff
@@ -6,3 +8,20 @@
 export function reform2005Applies(year: number): boolean {
     return year >= 2005;
 }
+
+/**
+* Helper function to check if settings have changed.
```
Comment (Log234) [inline src/utilities/conditionals.ts:13]: Codacy found an issue: [asterisks in jsdoc must be aligned](https://app.codacy.com/gh/Project-Lavinia/Lavinia-client/pullRequest?prid=6071614)

File: src/utilities/conditionals.ts [hunk 3]
```diff
@@ -6,3 +8,20 @@
...
+* @param settingsPayload The current chosen settings
+* @param comparison The original election settings
+*/
```
Comment (Log234) [inline src/utilities/conditionals.ts:16]: Codacy found an issue: [asterisks in jsdoc must be aligned](https://app.codacy.com/gh/Project-Lavinia/Lavinia-client/pullRequest?prid=6071614)

File: src/utilities/conditionals.ts [hunk 4]
```diff
@@ -6,3 +8,20 @@
 export function reform2005Applies(year: number): boolean {
     return year >= 2005;
 }
+
+/**
+* Helper function to check if settings have changed.
+* @param settingsPayload The current chosen settings
```
Comment (Log234) [inline src/utilities/conditionals.ts:14]: Codacy found an issue: [asterisks in jsdoc must be aligned](https://app.codacy.com/gh/Project-Lavinia/Lavinia-client/pullRequest?prid=6071614)

File: src/utilities/conditionals.ts [hunk 5]
```diff
@@ -6,3 +8,20 @@
...
+* @param comparison The original election settings
```
Comment (Log234) [inline src/utilities/conditionals.ts:15]: Codacy found an issue: [asterisks in jsdoc must be aligned](https://app.codacy.com/gh/Project-Lavinia/Lavinia-client/pullRequest?prid=6071614)

**Comment by Log234** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity decreasing per file
==============================
+ src/utilities/conditionals.ts  -1
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/Project-Lavinia/Lavinia-client/pullRequest?prid=6071614&bid=19881573)

**Comment by simendag** [comment]
This isn't really keeping track of whether settings changed; but whether they are different from the original, but I suppose it's better than how things are now. I don't particularly like us adding to the already quite bloated state, but I suppose there is no alternative.

**Review by simendag — APPROVED** [review]

**Comment by Log234** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity decreasing per file
==============================
+ src/utilities/conditionals.ts  -1
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/Project-Lavinia/Lavinia-client/pullRequest?prid=6071614&bid=19881573)

**Pull request merged**